src/Symfony/Component/Messenger/Tests/Transport/RedisExt/ConnectionTest.php

```diff
@@ -110,6 +110,17 @@ public function testAuth()
         Connection::fromDsn('redis://password@localhost/queue', [], $redis);
     }
 
+    public function testNoAuthWithEmptyPassword()
+    {
+        $redis = $this->getMockBuilder(\Redis::class)->disableOriginalConstructor()->getMock();
+
+        $redis->expects($this->exactly(0))->method('auth')
+            ->with('password')
+            ->willReturn(true);
+
+        Connection::fromDsn('redis://@localhost/queue', [], $redis);
```

Comment:
not sure we care, but you may test for `0@localhost` as well :)

Reply:
personally i would cast the empty string to null, where it applies. And let it bubble down as regular.

Reply:
Sounds good for me. Did add a test and cast to empty string to null.